### DISCUSSION TITLE
Adding worker nodes instructions

### DISCUSSION THREAD

**Discussion by hbjcr** [techno-tim/k3s-ansible#360] [Questions]
## Expected Behavior
Section with instructions explaining how to add new worker nodes

User should be able to follow the instructions to add a new node without having to rebuild the cluster from scratch

## Current Behavior

No instructions are currently provided

## Steps to Reproduce

NTR

## Context (variables)

Operating system:

Hardware: x64, arm64, armhf

## Possible Solution

- [ ] I've checked the [General Troubleshooting Guide](https://github.com/techno-tim/k3s-ansible/discussions/20)

**Comment by billsioros** [comment]
Could anyone who has already come across this issue please provide some instructions?

**Reply by emcniece** [reply]
https://github.com/techno-tim/k3s-ansible/discussions/360#discussioncomment-12111505

**Comment by fellnase** [comment]
This would be a nice feature, if I only have to add a new server or worker in the host.ini file and run the script again.
I also had the problem, that a node died and I have to setup it again. Because of the "special" configuration it is not so easy to know, how to do this manually.

**Reply by pbolduc** [reply]
If you wanted to do it manually, you could check out this [rpi4cluster](https://rpi4cluster.com/k3s/k3s-kube-setting/#workers). The author still uses ansible, but not this playbook.

**Reply by ruslanguns** [reply]
@pbolduc  the url provided is not working, could you check it please?

**Reply by pbolduc** [reply]
It is not my site. Seems the owner reorganized it. Just remove the path from the URL.

**Comment by emcniece** [comment]
This worked for me, Ubuntu 22.04:

0. Clone existing node
1. Edit `/etc/netplan/50-cloud-init.yaml`
    - Update IP address
    - Uptake new address: `netplan try`
2. Test `host google.ca` - if there are DNS errors, reboot
3. Edit K8s node annotations:
    - `k3s.io/internal-ip: [new ip]`
    - `k3s.io/node-args: ... --node-ip=[new ip]` flag
4. Edit `/etc/systemd/system/k3s-node.service`
    - Update `--node-ip=[new ip]` flag
    - systemctl daemon-reload
    - systemctl restart k3s-node

Step 3 may not be needed - I suspect these annotations might be generated during `k3s-node` service startup. Regardless it is useful to check those annotations after step 4 is complete.
